src/main/java/CliUI.java

```diff
@@ -0,0 +1,113 @@
+import java.util.Scanner;
+
+class CliUI {
+    private SnippetManager snippetManager;
+    private boolean isOpen;
+    private Scanner keyboard;
+
+    CliUI(SnippetManager snippetManager) {
+        this.snippetManager = snippetManager;
+        isOpen = true;
+        keyboard = new Scanner(System.in);
+    }
+
+    void uiLoop() {
+        while (isOpen) {
+            displayMenu();
+            getAndExecuteCommand();
+        }
+    }
+
+    private void displayMenu() {
+        System.out.println("Menu:\n1. View all snippets\n2. Create a snippet\n3. Delete a snippet\n4. Edit a snippet\n5. Filter\n6. Quit");
+    }
+
+    private void getAndExecuteCommand() {
+        int input = keyboard.nextInt();
+
+        switch (input) {
+            case 1: System.out.println(snippetManager.listAll()); break;
+            case 2: createSnippet(); break;
+            case 3: deleteSnippet(); break;
+            case 4: editSnippet(); break;
+            case 5: filterSnippets(); break;
+            case 6: isOpen = false; break;
+            default: System.out.println(input + ": please select a valid option");
+        }
+    }
```

Comment:
```suggestion
		     private void displayMenu() {
         System.out.println("Menu:\n1. View all snippets\n2. Create a snippet\n3. Delete a snippet\n4. Edit a snippet\n5. Filter\n6. Quit");
     }

      private void getAndExecuteCommand() {
         int input = keyboard.nextInt();

          switch (input) {
             case 1: System.out.println(snippetManager.listAll()); break;
             case 2: createSnippet(); break;
             case 3: deleteSnippet(); break;
             case 4: editSnippet(); break;
             case 5: filterSnippets(); break;
             case 6: isOpen = false; break;
             default: System.out.println(input + ": please select a valid option");
         }
     }
```
Good, compact function which is appropriately deemed private and delegates to multiple well-named functions. Nice!